sirepo/auth_db.py

```diff
@@ -140,6 +140,17 @@ def delete(self):
                 self._session().delete(self)
                 self._session().commit()
 
+        @classmethod
+        def delete_user(cls, uid):
+            """Delete user from all tables"""
+            for t in [x for x in cls.metadata.tables.values() if 'uid' in x.columns]:
```

Comment:
96cbb207d this changes to commit after updating all tables.

Reply:
That still won't work. Deletions have to be in order of constraints. For example, we might create the user_registration_t first, since that's the only table that provides a uniqueness constraint on the uid. We would then relate user_registration_t to auth_user_email_t on the field uid, which would ensure that auth_user_email_t had globally unique uids. The order of the tables is undefined in values() so it could be that auth_user_email_t was first (correct) for the delete or after user_registration_t (constraint violation).  

We should know the relationships and have a cascading delete. Barring that, we should list the tables explicitly with a reasonable order (user_registration_t last).  This type of code gets copy ("how do I delete all tables with a particular column in sqlalchemy?") so it's better not to have it as an example. If the table does not exist, then it should be checked against this values() list but otherwise not deleted. You could add an assertion if you like that all tables get visited.

Reply:
Thanks for the details. SQLite doesn't support `ADD CONSTRAINT` so I went with the the ordered list of tables. 46d3660dd

Reply:
I was thinking we'd be using a "real" DBMS when we added constraints. The point is that the code is hidden, and wouldn't obviously break.